Automattic/jetpack
# Connection: redirect users to proper Calypso environment during connection

## Proposed changes:
* Connection flow: redirect users to the Calypso environment chosen by developer instead of production.
* Add `calypsoEnv` into Connection package initial state, use it in `jp-redirect` component instead of 

### Other information:


## Jetpack product discussion
1204904295387809-as-1201303001696143

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Switch Calypso Env to `wpcalypso` (or something else) via `wp-config.php`: `define( 'CALYPSO_ENV', 'wpcalypso' );`
2. Try to connect Jetpack. Confirm that you got redirected to `https://wpcalypso.wordpress.com/` for approval. Complete the connection flow, confirm it works well, then disconnect.
3. Go to Jetpack Backup, click "Get VaultPress Backup". Confirm that you got redirected to the same `https://wpcalypso.wordpress.com` URL instead of `https://wordpress.com`. Complete the connection flow.
4. Go to "My Jetpack", scroll down to the "Purchase a plan" link, check the URL it leads to. There should be `calypso_env=wpcalypso` GET-parameter.
